src/App/Formatter/Post/CSV.php

```diff
@@ -67,26 +67,64 @@ public function generateHXLRows($header_rows, $hxl_rows)
         }
         return $hxl;
     }
-    private function generateHxlTagAndAttribute($key, $hxl_rows)
+
+    /**
+     * @param $form_attribute_key
+     * @param $hxl_rows
+     * @return string
+     * Iterates through all hxl attributes and looks for one that matches the given $form_attribute_key
+     */
+    private function generateHxlTagAndAttribute($form_attribute_key, $hxl_rows)
     {
         $tag = '';
         //TODO handle multi column values when CSV starts supporting them again
-
         foreach ($hxl_rows as $hxl_row) {
-            $key_set = explode('.', $key); //contains key + index of the key
-            $heading_key = isset($key_set[0]) ? $key_set[0] : null; // the heading type (sets, contact, title)
-            if ($hxl_row['key'] === $heading_key) {
-                if ($tag === '') {
-                    $tag = '#' . $hxl_row['tag_name'];
-                }
-                $attribute = $hxl_row['attribute'] && !empty($hxl_row['attribute']) ?
-                        ' +'. $hxl_row['attribute'] : '';
-                $tag = $tag.$attribute;
+            //contains key + index of the key
+            $key_set = explode('.', $form_attribute_key);
+            // the heading type (sets, contact, title)
+            $heading_key = isset($key_set[0]) ? $key_set[0] : null;
+
+            // checks that the hxl field matches the form attribute we received in $form_attribute_key
+            $key_matches_hxl = $hxl_row['key'] === $heading_key;
+            $should_process = $key_matches_hxl && $this->shouldProcessLatLonTag($key_set, $hxl_row);
+            if ($should_process) {
+                $tag = $this->returnHXLTagAndAttributeString($tag, $hxl_row);
             }
         }
         return $tag;
     }
 
+    private function shouldProcessLatLonTag($key_set, $hxl_row)
+    {
+        $should_not_process_lat_lon =
+            $hxl_row['tag_name'] !== 'geo' &&
+            $hxl_row['type'] !== 'point' &&
+            $hxl_row['input'] !== 'location';
+        if ($should_not_process_lat_lon) {
```

Comment:
I think the above can move into the if's conditional. I see the purpose of the var but we should use a comment instead if the conditional is complex.

Reply:
Alright. Changing now. I disagree as I prefer to read a conditional as a var instead of relying on a comment that may or may not be obsolete, but also see the other side. ..